homeassistant/components/ipma/__init__.py

```diff
@@ -57,4 +57,11 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> b
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
-    return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+    hass.data[DOMAIN].pop(entry.entry_id)
```

Comment:
```suggestion
    if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
        hass.data[DOMAIN].pop(entry.entry_id)

```
It is better not to remove the entry from `hass.data` in case of failure since it may still be running and using data

Reply:
Good catch, the variable was there for that purpose, but somehow missed it :(